statsmodels/graphics/regressionplots.py

```diff
@@ -406,15 +406,17 @@ def plot_partregress(endog, exog_i, exog_others, data=None,
     # all arrays or pandas-like
 
     if RHS_isemtpy:
+        endog = np.asarray(endog)
+        exog_i = np.asarray(exog_i)
         ax.plot(endog, exog_i, 'o', **kwargs)
         fitted_line = OLS(endog, exog_i).fit()
         x_axis_endog_name = 'x' if isinstance(exog_i, np.ndarray) else exog_i.name
         y_axis_endog_name = 'y' if isinstance(endog, np.ndarray) else endog.design_info.column_names[0]
```

Comment:
with the asarray above we always have np.ndarray and we loose the `names` which should be used in the plots.

Why do we need the asarray? What changed in pandas?